### PR TITLE
Fix: skip logging API keys in debug mode

### PR DESCRIPTION
**Pull request by Aswinr24**
Fixes issue #946

### CODE REVIEW

**Comment by codelion** [comment]
Instead of skipping such inputs can we just replace them with *s may be.

**Comment by Aswinr24** [comment]
Done !

**Comment by HarryBak** [comment]
It could be beneficial to treat this like credit card numbers.

Show some part, but mask most. Providing it's long enou6

**Comment by codelion** [comment]
This is a good idea, how about we show the first 4 and last 4 characters as long as the key is >= 24 characters. if it is less than 24 characters we mask with all stars. @Aswinr24

**Comment by Aswinr24** [comment]
Any more changes you suggest?

**Comment by codelion** [comment]
Is it okay to merge @CTY-git ?

File: patchwork/step.py [hunk 1]
```diff
@@ -130,6 +130,11 @@ def debug(self, inputs):
         MAX_LENGTH = 1000 # Max limit to print inputs
         printed_chars = 0
         for key, value in inputs.items():
+            if "api_key" in key.lower():
+                if len(value) >= 24:
+                    value = f"{value[:4]}{'*' * (len(value) - 8)}{value[-4:]}"
+                else:
+                    value = "*" * len(value)
```
Comment (CTY-git) [inline patchwork/step.py:137]: To be honest we can just return a constant string and disassociate any information about the value of the key from the censor/mask. I recommend just either `******` or `<masked api key>`.

**Review by CTY-git — CHANGES_REQUESTED** [review]

File: patchwork/step.py (hunk 1 above)
Comment (Aswinr24) [reply patchwork/step.py:137]: Okay seems like returning ```<masked api key>``` sounds better, Shall I go ahead with that then?

Comment (CTY-git) [reply patchwork/step.py:137]: Yes, please do. Thank you.

**Review by CTY-git — APPROVED** [review]

**Comment by CTY-git** [comment]
LGTM

**Pull request merged**